opm/material/components/CO2.hpp

```diff
@@ -247,6 +289,19 @@ class CO2 : public Component<Scalar, CO2<Scalar>>
         return (mu0 + dmu)/1.0e6; // conversion to [Pa s]
     }
 
+    /*!
+     * \brief The dynamic viscosity [Pa s] of CO2.
+     *
+     * This is only here so that the API is valid
```

Comment:
Lots of these functions added, can we not just remove them? What would be the consequence?

Reply:
removing these functions causes compilation errors, seems related to the common interface described in "Components.hpp", althought I am not sure why it does not just inherit that implementation which just throws an error

Reply:
And having added new versions of these that also take in the parameter objects it is a bit ugly as inheriting the interface from components does not make it obvious that you are supposed to create a CO2Tables object and send that in instead, not sure how to clean this up nicely